This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Boris Nagaev <bnagaev@gmail.com>
Date: Thu, 16 Mar 2017 04:09:42 +0100
Subject: [PATCH 1/2] fix out-of-tree build

Fix the error:
libotr-4.1.1/toolkit/otr_parse.c:26:19: fatal error: proto.h:
    No such file or directory

diff --git a/toolkit/Makefile.am b/toolkit/Makefile.am
index 1111111..2222222 100644
--- a/toolkit/Makefile.am
+++ b/toolkit/Makefile.am
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -I$(includedir) -I../src @LIBGCRYPT_CFLAGS@
+AM_CPPFLAGS = -I$(includedir) -I$(top_srcdir)/src @LIBGCRYPT_CFLAGS@
 
 noinst_HEADERS = aes.h ctrmode.h parse.h sesskeys.h readotr.h sha1hmac.h
 

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Boris Nagaev <bnagaev@gmail.com>
Date: Thu, 16 Mar 2017 06:09:50 +0100
Subject: [PATCH 2/2] .pc: add -fstack-protector-all and private libs

The flags taken from the build log.

diff --git a/libotr.pc.in b/libotr.pc.in
index 1111111..2222222 100644
--- a/libotr.pc.in
+++ b/libotr.pc.in
@@ -7,5 +7,6 @@ Name: libotr
 Description: Off-the-Record Messaging Library
 Version: @VERSION@
 URL: https://otr.cypherpunks.ca/
-Libs: -L${libdir} -lotr
+Libs: -L${libdir} -lotr -fstack-protector-all
+Libs.private: -lgcrypt -lgpg-error -lintl -liconv -lws2_32
 Cflags: -I${includedir}
